r2-lcp/src/main/java/org/readium/r2/lcp/service/LicensesRepository.kt

```diff
@@ -10,11 +10,32 @@
 package org.readium.r2.lcp.service
 
 import org.readium.r2.lcp.license.model.LicenseDocument
+import org.readium.r2.lcp.persistence.LcpDao
+import org.readium.r2.lcp.persistence.Licenses
 
-internal interface LicensesRepository {
-    fun addLicense(license: LicenseDocument)
-    fun copiesLeft(licenseId: String) : Int?
-    fun setCopiesLeft(quantity: Int, licenseId: String)
-    fun printsLeft(licenseId: String) : Int?
-    fun setPrintsLeft(quantity: Int, licenseId: String)
+class LicensesRepository(private val lcpDao: LcpDao) {
+
+    suspend fun addLicense(licenseDocument: LicenseDocument) {
+        if (lcpDao.exists(licenseDocument.id) != null) {
+            return
+        }
+        val license = Licenses(licenseDocument.id, licenseDocument.rights.print, licenseDocument.rights.copy)
```

Comment:
Since the arguments are a bunch of simple types, using named parameters is safer in case the order changes or is messed up.

```suggestion
        val license = License(licenseId = licenseDocument.id, rightPrint = licenseDocument.rights.print, rightCopy = licenseDocument.rights.copy)
```